src/main/scala/akka/contrib/process/NonBlockingProcess.scala

```diff
@@ -242,29 +242,55 @@ class NonBlockingProcess(
     pb.start()
   }
 
+  private var exitCodeToUse: Int = 255
+
   override def receive: Receive = {
+    case s: Started =>
+      context.parent ! s
+    case Exited(exitValue) =>
+      exitCodeToUse = exitValue
     case Destroy =>
       log.debug("Received request to destroy the process.")
       blocking(process.destroy(false))
     case DestroyForcibly =>
       log.debug("Received request to forcibly destroy the process.")
       blocking(process.destroy(true))
     case Inspect =>
-      val processIsRunning =
-        process.isRunning match {
-          // This additional check is needed for Linux given the lack of being able to
-          // detect based on closed file descriptors
-          case true if hasProcDir => procDir.resolve(process.getPID.toString).toFile.exists()
-          case result             => result
+      if (process.isRunning) {
+        // If NuProcess maintains we are running, we need to double-check as we could have died and children
+        // could still be alive and holding onto file descriptors.
+
+        procDirAlive(process.getPID).map { alive =>
```

Comment:
No test to see whether we should do a procdir test... I think it also reads better to have that condition tested here too.